ntoskrnl/kd64/kdprint.c

```diff
@@ -343,31 +343,22 @@ KdpPrint(
     STRING OutputString;
     CHAR CapturedString[512];
 
-    /* Assume failure */
-    *Handled = FALSE;
-
-#if (NTDDI_VERSION >= NTDDI_VISTA)
-    if ((ComponentId >= KdComponentTableSize) && (ComponentId < MAXULONG))
+    /* For such case, we just check the mask against system-wide one */
```

Comment:
Why is that now needed? It seems to break:
- both the original code (with the '||' replaced by '&&'),
- and what the NtQueryDebugFilterState() does?

Reply:
Looks to me like it replicates the previous behavior exactly. But the question is, whether this is desired and whether is makes any practical difference.

Reply:
This is mostly to replicate Windows 2003 behavior. I would rather move on and make it work like Vista+ - this is more practical (and maybe even don't take `Kd_WIN2000_Mask` into account because it looks useless when there is `Kd_DEFAULT_Mask`)

Reply:
The 2k3 behaviour should already be present even without this extra block, see the second commit (and the argumentation in the commit log).